horovod/run/common/util/safe_shell_exec.py

```diff
@@ -42,24 +44,32 @@ def terminate_executor_shell_and_children(pid):
             pass
 
     # Wait for graceful termination.
-    time.sleep(GRACEFUL_TERMINATION_TIME_S)
+    gone, alive = psutil.wait_procs(p.children(), timeout=GRACEFUL_TERMINATION_TIME_S)
 
     # Send STOP to executor shell to stop progress.
     p.send_signal(signal.SIGSTOP)
```

Comment:
Why `SIGSTOP`? stops a process that can be continued later. Is this about 'freezing' it so it does not spawn new children?

Reply:
I believe so.  @alsrgv may be able to comment further.

Reply:
STOP essentially freezes the process, it's not supposed to be scheduled to be executed until it's resumed.

Reply:
and the purpose of doing so in this method is? to avoid having it spawn new children? please add a comment to the code with the reason to STOP here.

Reply:
Yeah, that's exactly right.

Reply:
Updated comment.